packages/core/src/docdb/commands/renameInstance.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { getLogger, ToolkitError } from '../../shared'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { showViewLogsMessage } from '../../shared/utilities/messages'
+import { validateInstanceName } from '../utils'
+import { DBInstanceNode } from '../explorer/dbInstanceNode'
+import { telemetry } from '../../shared/telemetry'
+
+/**
+ * Renames a DocumentDB instance.
+ *
+ * Prompts the user for the new instance name
+ * Updates the instance.
+ * Refreshes the node.
+ */
+export async function renameInstance(node: DBInstanceNode) {
+    getLogger().debug('docdb:RenameInstance called for: %O', node)
+
+    await telemetry.docdb_renameInstance.run(async () => {
+        if (!node) {
+            throw new ToolkitError('No node specified for RenameInstance')
+        }
+
+        const instanceName = node.instance.DBInstanceIdentifier
+
+        if (node.instance.DBInstanceStatus !== 'available') {
```

Comment:
Same comment as earlier, I think I've seen this same code used ~6 or 7 times so far